pkg/kms/service.go

```diff
@@ -132,6 +134,26 @@ func (s *service) Decrypt(keystoreID, keyID string, cipher, aad, nonce []byte) (
 	return s.crypto.Decrypt(cipher, aad, nonce, kh)
 }
 
+// ComputeMAC computes message authentication code (MAC) for data.
+func (s *service) ComputeMAC(keystoreID, keyID string, data []byte) ([]byte, error) {
+	kh, err := s.getKeyHandle(keystoreID, keyID)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.crypto.ComputeMAC(data, kh)
+}
+
+// VerifyMAC determines if mac is a correct authentication code (MAC) for data.
+func (s *service) VerifyMAC(keystoreID, keyID string, mac, data []byte) error {
+	kh, err := s.getKeyHandle(keystoreID, keyID)
+	if err != nil {
+		return err
```

Comment:
Please wrap this error

Reply:
`getKeyHandle` is a helper function that already wraps an error, and also I am working on updating this logic in my next PR